### PR TITLE
Recursive query headers are added to the SQL dialects.

### PR DESCRIPTION
**Pull request by rita-gwen**
Two new methods added to simplify creation of recursive CTE queries in database-agnostic fashion. This should not affect any existing code.

### CODE REVIEW

**Comment by labkey-jeckels** [comment]
Matt, any comments on this?

File: api/src/org/labkey/api/data/dialect/SqlDialect.java [hunk 1]
```diff
@@ -1249,7 +1249,19 @@ public JdbcType getJdbcType(int type, String typeName)
         return t;
     }
 
-
+    /**
+     * Returns header of a recursive CTE query: WITH alias AS or WITH RECURSIVE alias AS
+     * depending on the SQL engine.
+     * @param tableAlias
```
Comment (labkey-jeckels) [inline api/src/org/labkey/api/data/dialect/SqlDialect.java]: Can you either remove these @param and @return lines, or add useful content?

Comment (rita-gwen) [reply api/src/org/labkey/api/data/dialect/SqlDialect.java]: Comments added.

Comment (labkey-jeckels) [reply api/src/org/labkey/api/data/dialect/SqlDialect.java]: Great, looks good from my side. Will wait for feedback from @labkey-matthewb as well.

**Comment by rita-gwen** [comment]
I am OK with adding the proposed methods to all the dialects, but writing proper unit tests looks beyond the scope of the current task at hand. That would require setting up and configuring databases for each dialect.
I prefer to roll this back and implement this logic in Panorama code. It will be easier since it supports PostgreSql and SqlServer only.
Thank you for the review.

**Pull request closed without merging**